doc/http_methods.rst

```diff
@@ -0,0 +1,221 @@
+
+    examplebuilder = DirectBuilder(metadata_file_contents, restrictions,"DELETE")
+
+On calling *SingleQueryable.generate()*, this would return a fuzzed URI string and an empty JSON string.
+
+::
+     
+     Categories?sap-client=500
+
+
+
+POST Queries
+------------
+
+For generating POST queries, DirectBuilder needs to be called with **method = "POST"**
+
+.. code-block:: python
+
+    examplebuilder = DirectBuilder(metadata_file_contents, restrictions,"GET")
+
+On calling *SingleQueryable.generate()*, this would return a fuzzed URI string and a JSON string containing all the fuzzed properties. No keys appear in the URI.
+
+::
+     
+     Categories?sap-client=500
+     
+     {"CategoryID": "-346633563", "CategoryName": "DNf%C2%90", "Description": "%E2%80%93Qe%C3%94%3C2%C3%B9%C3%9F%2A%C2%AC%E2%84%A2%C3%BB%C3%86E6m%40%C3%A5%C2%BA%C3%BB%C2%A9%C2%B9o1%C3%94%C2%90%C2%AAA%C2%A9%C3%A5A%E2%80%A2%C2%AC%20%C3%92%C2%BB%C2%A2%C2%B0%C3%96h%C2%8D%C3%BF%C5%92%C3%85u%3C", "Picture": "YmluYXJ5JzcyJw=="}
+
+
+PUT Queries
+-----------
+
+For generating PUT queries, DirectBuilder needs to be called with **method = "PUT"**
+
+.. code-block:: python
+
+    examplebuilder = DirectBuilder(metadata_file_contents, restrictions,"PUT")
+
+On calling *SingleQueryable.generate()*, this would return a fuzzed URI string and a JSON string containing all the fuzzed properties. All the keys appear in the URI and are synchronized with the body. 
+
+::
+     
+     Categories(CategoryID=1714953551,CategoryName='%21%C2%9Dla%C3%92l%24',Description='hz%60%C3%8F%C3%8F%7B%C3%AAi%2Bk%C3%81%C2%A4%C3%96xc%C5%93%C2%A85k%C3%93%2A%C3%B5%C2%BBrLD%2A%E2%80%A1',Picture=binary'ac9916669fAeb2')?sap-client=500   
+     
+     {"CategoryID": "1714953551", "CategoryName": "%21%C2%9Dla%C3%92l%24", "Description": "hz%60%C3%8F%C3%8F%7B%C3%AAi%2Bk%C3%81%C2%A4%C3%96xc%C5%93%C2%A85k%C3%93%2A%C3%B5%C2%BBrLD%2A%E2%80%A1", "Picture": "YmluYXJ5J2FjOTkxNjY2OWZBZWIyJw=="}
+
+
+
+
+
+==================
+Code Documentation
+==================
+
+Adding method parameter in DirectBuilder
+----------------------------------------
+
+DirectBuilder now has an additional parameter called **method**. This accepts the users choice of HTTP method to get the fuzzed requests. The DirectBuilder *init* checks for validity and calls the *Config.fuzzer* setter to set the value in the Config object.
+
+.. code-block:: python
+
+    def __init__(self, metadata, restrictions,method):
+        if method not in ["GET","DELETE","PUT","POST"]:
+            raise ValueError("The http method value \'{}\' is invalid\nUse either GET, DELETE, PUT or POST".format(method))
+        self._queryable = QueryableEntities()
+        self._metadata_string = metadata
+        self._restrictions = restrictions
+        Config.init()
+        Config.fuzzer.http_method_enabled = method
+
+In config.py, this would be used to build the Config object, which would be looked up during query construction and check which HTTP method is set.
```

Comment:
This block should be on the top of the document, after Introduction section, since after enumerating all the possible methods and how to call them, servers no purpose. On the contrary, before diving to differences between HTTP methods, this is introduction to the class usage and it is generic for all http methods.